packages/mui-material/src/Select/Select.spec.tsx

```diff
@@ -75,4 +75,127 @@ function genericValueTest() {
       },
     }}
   />;
+
+  <Select variant="filled" />;
+  <Select variant="standard" />;
+  <Select variant="outlined" />;
+  <Select />;
+
+  <Select variant="filled" hiddenLabel />;
+  // @ts-expect-error hiddenLabel is not present in standard variant
+  <Select variant="standard" hiddenLabel />;
+  // @ts-expect-error hiddenLabel is not present in outlined variant
+  <Select variant="outlined" hiddenLabel />;
+  // @ts-expect-error hiddenLabel is not present in outlined variant
+  <Select hiddenLabel />;
+
+  const defaultProps: SelectProps<number> = {};
+  const outlinedProps: SelectProps<number> = {
+    variant: 'outlined',
+  };
+  const filledProps: SelectProps<number> = {
+    variant: 'filled',
+  };
+  const standardProps: SelectProps<number> = {
+    variant: 'standard',
+  };
+
+  <Select {...defaultProps} />;
+  <Select {...outlinedProps} />;
+  <Select {...filledProps} />;
+  <Select {...standardProps} />;
+  <Select<number> {...outlinedProps} />;
+  <Select<number> {...defaultProps} />;
+  <Select<number> {...filledProps} />;
+
+  const rawDefaultProps: SelectProps = {};
+  const rawOutlinedProps: SelectProps = {
+    variant: 'outlined',
+  };
+  const rawFilledProps: SelectProps = {
+    variant: 'filled',
+  };
+
+  <Select {...rawDefaultProps} />;
+  <Select {...rawOutlinedProps} />;
+  <Select {...rawFilledProps} />;
+
+  // @ts-expect-error hiddenLabel is not present in outlined variant
+  <Select {...defaultProps} hiddenLabel />;
+  // @ts-expect-error hiddenLabel is not present in outlined variant
+  <Select {...outlinedProps} hiddenLabel />;
+  <Select {...filledProps} hiddenLabel />;
+  // @ts-expect-error hiddenLabel is not present in standard variant
+  <Select {...standardProps} hiddenLabel />;
+
+  interface OtherProps {
+    otherProp?: number;
+  }
+
+  const SelectWrapper1 = <Value = unknown,>(props: SelectProps<Value> & OtherProps) => {
```

Comment:
We need the comma for the parser to recognize it as a TypeScript type rather than a JSX tag, but we can remove the default value:
```suggestion
  const SelectWrapper1 = <Value,>(props: SelectProps<Value> & OtherProps) => {
```